pypesto/objective/amici_calculator.py

```diff
@@ -89,14 +92,31 @@ def __call__(self,
             parameter_mapping=parameter_mapping,
             amici_model=amici_model
         )
+        rdatas = []
+        # spliting the ExpData in chunks and simulating chunk wise
+        # Do we have a maximum number of simulations allowed?
+        n_edatas = len(edatas)
+        if chunk_size is None:
+            # simulate all conditions at once
+            n_simulations = 1
+        else:
+            # simulate only a subset of conditions
+            n_simulations = int(np.ceil(len(edatas)/chunk_size))
+        for i_sim in range(n_simulations):
+            # slice out the conditions we actually want
+            if chunk_size is None:
+                ids = slice(0, n_edatas)
+            else:
+                ids = slice(i_sim * chunk_size,
+                            min((i_sim + 1) * chunk_size, n_edatas))
+            # run amici simulation
+            rdatas.extend(amici.runAmiciSimulations(
+                amici_model,
+                amici_solver,
+                edatas[ids],
+                num_threads=min(n_threads, len(edatas)),
+            ))
```

Comment:
Wait, this won't work. You *need* something like 
```
if mode == mode_fun:
    requested_keys = ['llh', 'sllh', 'status']
elif  mode == mode_res:
     requested_keys = ['res', 'sres', 'status']
 else:
     requested_keys = None
chunks = amici.runAmiciSimulaitons()...,
if requested_keys is not None:
    rdata.extend([{key: chunk[key] for key in requested keys} for chunk in chunks])
else:
    ...your code...
```

And that thing should in addition be plugged into a separate function. Then, variable scoping will ensure that "chunk" is indeed deleted...

Reply:
ok, that's what I thought about as well, as we are always passing on the data object :+1: will make changes accordingly.